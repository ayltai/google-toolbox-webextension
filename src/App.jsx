import { createMuiTheme, ThemeProvider, useMediaQuery, } from '@material-ui/core';
import arrayMove from 'array-move';
import React from 'react';
import browser from 'webextension-polyfill';

import { DroppableGridList, } from './components/DroppableGridList';
import { CONFIGURATIONS, } from './Config';
import { SERVICES, } from './Data';

export const App = () => {
    const isDarkMode = useMediaQuery('(prefers-color-scheme: dark)');

    const theme = React.useMemo(() => createMuiTheme({
        palette : {
            type : isDarkMode ? 'dark' : 'light',
        },
    }), [ isDarkMode, ]);

    const [ state, setState, ] = React.useState({
        dragged  : false,
        services : SERVICES,
    });

    if (!state.configuration) browser.storage.sync.get({
        numRows      : CONFIGURATIONS.medium.rows,
        compactStyle : false,
    }, preferences => {
        const configuration = preferences.compactStyle ? CONFIGURATIONS.small : CONFIGURATIONS.medium;
        configuration.rows = preferences.numRows;

        setState({
            ...state,
            configuration,
        });
    });

    return (
        <ThemeProvider theme={theme}>
            <DroppableGridList
                axis='xy'
                services={state.services}
                configuration={state.configuration || CONFIGURATIONS.medium}
                onSortStart={(props, event) => event.preventDefault()}
                onSortMove={() => {
                    setState({
                        ...state,
                        dragged : true,
                    });
                }}
                onSortEnd={({ oldIndex, newIndex, }) => {
                    if (oldIndex === newIndex) {
                        if (state.dragged) {
                            setState({
                                ...state,
                                dragged : false,
                            });
                        } else {
                            browser.tabs.create({
                                url : state.services[oldIndex].url,
                            });

                            window.close();
                        }
                    } else {
                        setState({
                            ...state,
                            dragged  : false,
                            services : arrayMove(state.services, oldIndex, newIndex),
                        });
                    }
                }} />
        </ThemeProvider>
    );
};
