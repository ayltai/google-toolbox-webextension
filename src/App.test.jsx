import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import browser from 'sinon-chrome/webextensions';

import { App } from './App';

window.browser = browser;

describe('<App />', () => {
    it('renders correctly', () => {
        expect(renderer.create(<App />).toJSON()).toMatchSnapshot();
    });

    it('renders without errors', () => {
        const div = document.createElement('div');

        ReactDOM.render(<App />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
});
