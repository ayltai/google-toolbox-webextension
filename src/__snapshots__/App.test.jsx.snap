// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div
  className="makeStyles-container-2"
>
  <ul
    className="MuiGridList-root"
    style={
      Object {
        "margin": -2,
      }
    }
  >
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <button
          aria-describedby={null}
          className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-6 MuiIconButton-sizeSmall"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="Docs"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <div
              className="MuiAvatar-root MuiAvatar-square makeStyles-avatar-7"
            >
              <img
                className="MuiAvatar-img"
                src="images/docs.png"
              />
            </div>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <button
          aria-describedby={null}
          className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-42 MuiIconButton-sizeSmall"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="Drive"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <div
              className="MuiAvatar-root MuiAvatar-square makeStyles-avatar-43"
            >
              <img
                className="MuiAvatar-img"
                src="images/drive.png"
              />
            </div>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <button
          aria-describedby={null}
          className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-44 MuiIconButton-sizeSmall"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="Gmail"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <div
              className="MuiAvatar-root MuiAvatar-square makeStyles-avatar-45"
            >
              <img
                className="MuiAvatar-img"
                src="images/gmail.png"
              />
            </div>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <button
          aria-describedby={null}
          className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-46 MuiIconButton-sizeSmall"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="Photos"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <div
              className="MuiAvatar-root MuiAvatar-square makeStyles-avatar-47"
            >
              <img
                className="MuiAvatar-img"
                src="images/photos.png"
              />
            </div>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <button
          aria-describedby={null}
          className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-48 MuiIconButton-sizeSmall"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="Maps"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <div
              className="MuiAvatar-root MuiAvatar-square makeStyles-avatar-49"
            >
              <img
                className="MuiAvatar-img"
                src="images/maps.png"
              />
            </div>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <button
          aria-describedby={null}
          className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-50 MuiIconButton-sizeSmall"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="Sheets"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <div
              className="MuiAvatar-root MuiAvatar-square makeStyles-avatar-51"
            >
              <img
                className="MuiAvatar-img"
                src="images/sheets.png"
              />
            </div>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <button
          aria-describedby={null}
          className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-52 MuiIconButton-sizeSmall"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="Slides"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <div
              className="MuiAvatar-root MuiAvatar-square makeStyles-avatar-53"
            >
              <img
                className="MuiAvatar-img"
                src="images/slides.png"
              />
            </div>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <button
          aria-describedby={null}
          className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-54 MuiIconButton-sizeSmall"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="YouTube"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <div
              className="MuiAvatar-root MuiAvatar-square makeStyles-avatar-55"
            >
              <img
                className="MuiAvatar-img"
                src="images/youtube.png"
              />
            </div>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </li>
  </ul>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-4"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <span
        className="MuiButton-startIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </span>
      More
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;
